platform/mellanox/mlnx-platform-api/sonic_platform/psu.py

```diff
@@ -505,6 +512,55 @@ def get_input_current(self):
             return float(amperes) / 1000
         return None
 
+    def _get_psu_power_threshold(self, temp_threshold_path):
+        """
+        Calculate power threshold for a PSU according to the maximum power capacity and ambient temperature
+            amb_temp = min(port_amb, fan_amb)
+            If amb_temp < ambient_temp_threshold
+                threshold = max capacity
+            else
+                threshold = max capacity - slope*(amb_temp - ambient_temp_threshold)
+        """
+        if self.get_powergood_status():
+            if os.path.exists(self.psu_power_max_capacity):
+                power_max_capacity = utils.read_int_from_file(self.psu_power_max_capacity)
+                temp_threshold = utils.read_int_from_file(temp_threshold_path)
+                fan_ambient_temp = utils.read_int_from_file(Psu.FAN_AMBIENT_TEMP)
+                port_ambient_temp = utils.read_int_from_file(Psu.PORT_AMBIENT_TEMP)
+                ambient_temp = min(fan_ambient_temp, port_ambient_temp)
+                if ambient_temp < temp_threshold:
+                    power_threshold = power_max_capacity
+                else:
+                    slope = utils.read_int_from_file(Psu.PSU_POWER_SLOPE)
+                    power_threshold = power_max_capacity - (ambient_temp - temp_threshold) * slope
+                if power_threshold <= 0:
+                    logger.log_warning('Got negative PSU power threshold {} for {}'.format(power_threshold, self.get_name()))
+                    power_threshold = 0
+                return float(power_threshold) / 1000000
+
+        return None
+
+    def get_psu_power_warning_threshold(self):
```

Comment:
can we rename as discuss in the HLD?

Reply:
Done. Changed to `get_psu_power_warning_suppress_threshold`

Reply:
@stephenxs description still has warning

Reply:
You were referring to an old commit...